### ISSUE TITLE
blade fails to build on FreeBSD

### ISSUE THREAD

**Issue by yurivict** [kvark/blade#89]
```
   Compiling blade-graphics v0.3.0 (/usr/ports/editors/zed/blade/blade-graphics)
error[E0583]: file not found for module `hal`
  --> blade-graphics/src/lib.rs:62:1
   |
62 | mod hal;
   | ^^^^^^^^
   |
   = help: to create the module `hal`, create file "blade-graphics/src/hal.rs" or "blade-graphics/src/hal/mod.rs"
   = note: if there is a `mod hal` elsewhere in the crate already, import it with `use crate::...` instead

error[E0412]: cannot find type `Context` in module `super`
  --> blade-graphics/src/shader.rs:11:13
   |
11 | impl super::Context {
   |             ^^^^^^^ not found in `super`
   |
help: consider importing one of these items
   |
1  + use core::task::Context;
   |
1  + use std::task::Context;
   |
help: if you import `Context`, refer to it directly
   |
11 - impl super::Context {
11 + impl Context {
   |

error[E0412]: cannot find type `AccelerationStructure` in the crate root
   --> blade-graphics/src/traits.rs:70:40
    |
70  |         acceleration_structure: crate::AccelerationStructure,
    |                                        ^^^^^^^^^^^^^^^^^^^^^
    |
   ::: blade-graphics/src/lib.rs:449:1
    |
449 | pub struct AccelerationStructureDesc<'a> {
    | ---------------------------------------- similarly named struct `AccelerationStructureDesc` defined here
    |
help: there is an enum variant `crate::ShaderBinding::AccelerationStructure` and 1 other; try using the variant's enum
    |
70  |         acceleration_structure: crate::ShaderBinding,
    |                                 ~~~~~~~~~~~~~~~~~~~~
70  |         acceleration_structure: naga::TypeInner,
    |                                 ~~~~~~~~~~~~~~~
help: a struct with a similar name exists
    |
70  |         acceleration_structure: crate::AccelerationStructureDesc,
    |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0412]: cannot find type `AccelerationStructure` in the crate root
   --> blade-graphics/src/traits.rs:77:40
    |
77  |         acceleration_structure: crate::AccelerationStructure,
    |                                        ^^^^^^^^^^^^^^^^^^^^^
    |
   ::: blade-graphics/src/lib.rs:449:1
    |
449 | pub struct AccelerationStructureDesc<'a> {
    | ---------------------------------------- similarly named struct `AccelerationStructureDesc` defined here
    |
help: there is an enum variant `crate::ShaderBinding::AccelerationStructure` and 1 other; try using the variant's enum
    |
77  |         acceleration_structure: crate::ShaderBinding,
    |                                 ~~~~~~~~~~~~~~~~~~~~
77  |         acceleration_structure: naga::TypeInner,
    |                                 ~~~~~~~~~~~~~~~
help: a struct with a similar name exists
    |
77  |         acceleration_structure: crate::AccelerationStructureDesc,
    |                                        ~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by kvark** [comment]
Thank you for filing! If you tweak the compile time configurations from `linux` to `linux | bsd`, and blade-graphics works (at least with GLES backend), that would be great to get in!

**Comment by kvark** [comment]
Fixed by #152
